astropy/cosmology/tests/test_core.py

```diff
@@ -74,9 +74,13 @@ def __init__(self, H0, Tcmb0=0*u.K, name=None, meta=None):
                 self._Tcmb0 = Tcmb0
 
         self.cls = SubCosmology
-        self.cls_args = (70 * (u.km / u.s / u.Mpc), 2.7 * u.K)
+        self._cls_args = dict(H0=70 * (u.km / u.s / u.Mpc), Tcmb0=2.7 * u.K)
         self.cls_kwargs = dict(name=self.__class__.__name__, meta={"a": "b"})
 
+    @property
```

Comment:
Would it not make more sense to just change `cls_args` to a dict and adjust the tests? As long as you have not advertised the testing machinery, I'd consider it private implementation details.

Reply:
I have not yet advertised the testing machinery, so that's good.
I'll definitely think about changing it. The current setup allows for `*args`, which a dict would not. But there may be more elegant solutions. There's some time before v5.1, when I aim to make this public.

Reply:
And thanks, as always for the reviews